### PR TITLE
chore: add CODEOWNERS

### PR DESCRIPTION
**Pull request by Boshen**
## Summary

Setup CODEOWNERS for easier reviewer assignments.
@hardfist Can you assign these to the correct person, you can push to this branch.

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [x] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run changeset`.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/crates/cargo-rst @IWANABETHATGUY
+/crates/loader_runner
+/crates/node_binding
+/crates/rspack
+/crates/rspack_ast_viewer
+/crates/rspack_base64
+/crates/rspack_binding_macros
+/crates/rspack_binding_options
+/crates/rspack_build
+/crates/rspack_core
```
Comment (Boshen) [inline .github/CODEOWNERS:30]: core can be split up.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# Examples
+/examples
+
+# Crates in alphabetical order
+/crates/bench @IWANABETHATGUY
+/crates/cargo-rst @IWANABETHATGUY
+/crates/loader_runner
+/crates/node_binding
+/crates/rspack
+/crates/rspack_ast_viewer
```
Comment (IWANABETHATGUY) [inline .github/CODEOWNERS:25]: I am pretty sure this belong to @h-a-n-a

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+/benchcases @IWANABETHATGUY
+/xtask @IWANABETHATGUY
+/.github/workflows @IWANABETHATGUY @Boshen
+
+# Examples
+/examples
+
+# Crates in alphabetical order
+/crates/bench @IWANABETHATGUY
+/crates/cargo-rst @IWANABETHATGUY
```
Comment (IWANABETHATGUY) [inline .github/CODEOWNERS:21]: @JSerFeng create cargo-rst, so please add him

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_fs_node
+/crates/rspack_futures
+/crates/rspack_identifier
+/crates/rspack_ids
+/crates/rspack_loader_sass
+/crates/rspack_napi_shared
+/crates/rspack_plugin_asset
+/crates/rspack_plugin_copy
+/crates/rspack_plugin_css
+/crates/rspack_plugin_dev_friendly_split_chunks
```
Comment (hyf0) [inline .github/CODEOWNERS:45]: me

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_plugin_devtool
+/crates/rspack_plugin_externals
+/crates/rspack_plugin_html
+/crates/rspack_plugin_javascript
+/crates/rspack_plugin_json
+/crates/rspack_plugin_library
+/crates/rspack_plugin_progress
+/crates/rspack_plugin_remove_empty_chunks
```
Comment (hyf0) [inline .github/CODEOWNERS:53]: me

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_plugin_runtime
+/crates/rspack_plugin_split_chunks
```
Comment (hyf0) [inline .github/CODEOWNERS:55]: me

File: .github/CODEOWNERS [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack
+/crates/rspack_ast_viewer @h-a-n-a
+/crates/rspack_base64
+/crates/rspack_binding_macros
+/crates/rspack_binding_options
+/crates/rspack_build
+/crates/rspack_core/src/build_chunk_graph @hyf0
+/crates/rspack_core/src/tree_shaking @IWANABETHATGUY
+/crates/rspack_core
+/crates/rspack_database
```
Comment (hyf0) [inline .github/CODEOWNERS:33]: me

File: .github/CODEOWNERS [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_error
+/crates/rspack_fs
+/crates/rspack_fs_node
+/crates/rspack_futures
+/crates/rspack_identifier
+/crates/rspack_ids
```
Comment (hyf0) [inline .github/CODEOWNERS:39]: me

File: .github/CODEOWNERS [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_plugin_javascript
+/crates/rspack_plugin_json
+/crates/rspack_plugin_library
+/crates/rspack_plugin_progress
+/crates/rspack_plugin_remove_empty_chunks
+/crates/rspack_plugin_runtime
+/crates/rspack_plugin_split_chunks
+/crates/rspack_plugin_wasm
+/crates/rspack_regex
+/crates/rspack_symbol
```
Comment (IWANABETHATGUY) [inline .github/CODEOWNERS:58]: me

File: .github/CODEOWNERS [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_plugin_html
+/crates/rspack_plugin_javascript
+/crates/rspack_plugin_json
+/crates/rspack_plugin_library
+/crates/rspack_plugin_progress
+/crates/rspack_plugin_remove_empty_chunks
+/crates/rspack_plugin_runtime
+/crates/rspack_plugin_split_chunks
+/crates/rspack_plugin_wasm
+/crates/rspack_regex
```
Comment (IWANABETHATGUY) [inline .github/CODEOWNERS:57]: me

File: .github/CODEOWNERS [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_core/src/tree_shaking @IWANABETHATGUY
+/crates/rspack_core
+/crates/rspack_database
+/crates/rspack_error
+/crates/rspack_fs
+/crates/rspack_fs_node
+/crates/rspack_futures
+/crates/rspack_identifier
+/crates/rspack_ids
+/crates/rspack_loader_sass
```
Comment (ahabhgk) [inline .github/CODEOWNERS:40]: me

File: .github/CODEOWNERS [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_napi_shared
+/crates/rspack_plugin_asset
```
Comment (ahabhgk) [inline .github/CODEOWNERS:42]: me

File: .github/CODEOWNERS [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_plugin_copy
+/crates/rspack_plugin_css
```
Comment (ahabhgk) [inline .github/CODEOWNERS:44]: me

File: .github/CODEOWNERS [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_plugin_dev_friendly_split_chunks
+/crates/rspack_plugin_devtool
```
Comment (ahabhgk) [inline .github/CODEOWNERS:46]: me

File: .github/CODEOWNERS [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_plugin_externals
+/crates/rspack_plugin_html
+/crates/rspack_plugin_javascript
+/crates/rspack_plugin_json
```
Comment (ahabhgk) [inline .github/CODEOWNERS:50]: me

File: .github/CODEOWNERS [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_plugin_library
+/crates/rspack_plugin_progress
+/crates/rspack_plugin_remove_empty_chunks
+/crates/rspack_plugin_runtime
+/crates/rspack_plugin_split_chunks
+/crates/rspack_plugin_wasm
```
Comment (ahabhgk) [inline .github/CODEOWNERS:56]: me

File: .github/CODEOWNERS [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+# Packages in alphabetical order
+/packages/create-rspack
+/packages/less-loader
+/packages/postcss-loader
+/packages/rspack
+/packages/rspack-cli
+/packages/rspack-dev-client
+/packages/rspack-dev-middleware
+/packages/rspack-dev-server
+/packages/rspack-plugin-html
```
Comment (ahabhgk) [inline .github/CODEOWNERS:72]: me

File: .github/CODEOWNERS [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_identifier
+/crates/rspack_ids
+/crates/rspack_loader_sass
+/crates/rspack_napi_shared
+/crates/rspack_plugin_asset
+/crates/rspack_plugin_copy
+/crates/rspack_plugin_css
+/crates/rspack_plugin_dev_friendly_split_chunks
+/crates/rspack_plugin_devtool
+/crates/rspack_plugin_externals
```
Comment (ahabhgk) [inline .github/CODEOWNERS:47]: me

File: .github/CODEOWNERS [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_plugin_html
```
Comment (ahabhgk) [inline .github/CODEOWNERS:48]: me

**Comment by IWANABETHATGUY** [comment]
@web-infra-dev/rspack

File: .github/CODEOWNERS (hunk 6 above)
Comment (h-a-n-a) [reply .github/CODEOWNERS:55]: +1

File: .github/CODEOWNERS (hunk 12 above)
Comment (h-a-n-a) [reply .github/CODEOWNERS:42]: +1

File: .github/CODEOWNERS (hunk 15 above)
Comment (h-a-n-a) [reply .github/CODEOWNERS:50]: +1

File: .github/CODEOWNERS (hunk 16 above)
Comment (h-a-n-a) [reply .github/CODEOWNERS:56]: +1

File: .github/CODEOWNERS (hunk 18 above)
Comment (h-a-n-a) [reply .github/CODEOWNERS:47]: +1

File: .github/CODEOWNERS [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+/.github/workflows @IWANABETHATGUY @Boshen
+
+# Examples
+/examples
+
+# Crates in alphabetical order
+/crates/bench @IWANABETHATGUY
+/crates/cargo-rst @JSerFeng
+/crates/loader_runner
+/crates/node_binding
```
Comment (h-a-n-a) [inline .github/CODEOWNERS:23]: +1

File: .github/CODEOWNERS [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+* @hardfist
+
+# Ignore lock files
+Cargo.lock
+pnpm-lock.yaml
+
+# Infra
+/benchcases @IWANABETHATGUY
+/xtask @IWANABETHATGUY
+/.github/workflows @IWANABETHATGUY @Boshen
```
Comment (h-a-n-a) [inline .github/CODEOWNERS:14]: +1

File: .github/CODEOWNERS [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+/examples
+
+# Crates in alphabetical order
+/crates/bench @IWANABETHATGUY
+/crates/cargo-rst @JSerFeng
+/crates/loader_runner
+/crates/node_binding
+/crates/rspack
+/crates/rspack_ast_viewer @h-a-n-a
+/crates/rspack_base64
```
Comment (h-a-n-a) [inline .github/CODEOWNERS:26]: +1

File: .github/CODEOWNERS [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_binding_macros
```
Comment (h-a-n-a) [inline .github/CODEOWNERS:27]: +1

File: .github/CODEOWNERS [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_binding_options
```
Comment (h-a-n-a) [inline .github/CODEOWNERS:28]: +1

File: .github/CODEOWNERS [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_build
+/crates/rspack_core/src/build_chunk_graph @hyf0
+/crates/rspack_core/src/tree_shaking @IWANABETHATGUY
+/crates/rspack_core
+/crates/rspack_database
+/crates/rspack_error
```
Comment (h-a-n-a) [inline .github/CODEOWNERS:34]: +1

File: .github/CODEOWNERS [hunk 26]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_fs
+/crates/rspack_fs_node
+/crates/rspack_futures
+/crates/rspack_identifier
```
Comment (h-a-n-a) [inline .github/CODEOWNERS:38]: +1

File: .github/CODEOWNERS [hunk 27]
```diff
@@ -0,0 +1,74 @@
...
+/crates/rspack_ids
+/crates/rspack_loader_sass
+/crates/rspack_napi_shared
```
Comment (h-a-n-a) [inline .github/CODEOWNERS:41]: +1

**Comment by h-a-n-a** [comment]
@web-infra-dev/rspack

**Comment by Boshen** [comment]
@web-infra-dev/rspack Can you double check and assign the node packages as well?

**Comment by Boshen** [comment]
@hardfist can you double check?

File: .github/CODEOWNERS [hunk 28]
```diff
@@ -0,0 +1,73 @@
...
+# Packages in alphabetical order
+/packages/create-rspack
+/packages/postcss-loader
+/packages/rspack
+/packages/rspack-cli
+/packages/rspack-dev-client
+/packages/rspack-dev-middleware
+/packages/rspack-dev-server
+/packages/rspack-plugin-html @ahabhgk
+/packages/rspack-plugin-minify
```
Comment (hardfist) [inline .github/CODEOWNERS:72]: me

File: .github/CODEOWNERS [hunk 29]
```diff
@@ -0,0 +1,73 @@
...
+/crates/rspack_regex @IWANABETHATGUY
+/crates/rspack_symbol @IWANABETHATGUY
+/crates/rspack_testing @ahabhgk
+/crates/rspack_tracing @hyf0
+/crates/rspack_util @hyf0
+
+# Packages in alphabetical order
+/packages/create-rspack
+/packages/postcss-loader
+/packages/rspack
```
Comment (hardfist) [inline .github/CODEOWNERS:66]: me

File: .github/CODEOWNERS [hunk 30]
```diff
@@ -0,0 +1,73 @@
...
+/crates/rspack_plugin_split_chunks @hyf0 @h-a-n-a
+/crates/rspack_plugin_wasm @ahabhgk @h-a-n-a
+/crates/rspack_regex @IWANABETHATGUY
+/crates/rspack_symbol @IWANABETHATGUY
+/crates/rspack_testing @ahabhgk
+/crates/rspack_tracing @hyf0
+/crates/rspack_util @hyf0
+
+# Packages in alphabetical order
+/packages/create-rspack
```
Comment (hardfist) [inline .github/CODEOWNERS:64]: me

File: .github/CODEOWNERS [hunk 31]
```diff
@@ -0,0 +1,73 @@
...
+/packages/postcss-loader
```
Comment (hardfist) [inline .github/CODEOWNERS:65]: @ahabhgk

File: .github/CODEOWNERS [hunk 32]
```diff
@@ -0,0 +1,73 @@
...
+Cargo.lock
+pnpm-lock.yaml
+
+# Infra
+/benchcases @IWANABETHATGUY
+/xtask @IWANABETHATGUY
+/.github/workflows @IWANABETHATGUY @h-a-n-a
+
+# Examples
+/examples
```
Comment (hardfist) [inline .github/CODEOWNERS:17]: me

**Review by hardfist — APPROVED** [review]

**Pull request merged**